python/client.py

```diff
@@ -36,7 +36,14 @@ def save_message(msg: ServiceBusReceivedMessage, download_directory):
     raw_json_s = ast.literal_eval(str(msg))
     raw_json = json.loads(raw_json_s)
     dataset = msg.subject or 'unknown'
-    file_name = f'{dataset}_{datetime.now().strftime("%y%m%dT%H%M%S_%f")}.json'
+
+    properties = msg.application_properties or {}
+    original_file_name = properties.get(b"source_file_name", b"UNKNOWN").decode("utf-8")
```

Comment:
Is this the same property as `messageId`  in the [other PR](https://github.com/elexon-data/iris-clients/pull/20/files#diff-acd2903eb3e768df2a0d683907d6512fcb651ec00baa279acdd62472c5e0e7fc)?

Reply:
Ah the DevOps ticket has some record of this - 
"the iris-testing-client currently saves messages with filename = source_file_name, and the default iris clients construct their own filename
there's already [a pull request from Ying](https://github.com/elexon-data/iris-clients/pull/20) that makes this change for the nodejs and .NET clients
there's already a direct branch from Becky that makes this change for python clients, but uses the property source_file_name instead of message_id, to match the iris-testing-client
upon discussing this we have agreed to use message_id everywhere, because this is definitely unique, whereas source_file_name could be non-unique if a dataset is split up by source file e.g. SAA datasets"

Reply:
![image](https://github.com/elexon-data/iris-clients/assets/112951700/699f0432-6ed6-4d53-9d1d-6bc4c735ad20)
Looks like not the same property, but the same value. Which one would we prefer to use? Happy to change either PR to match